src/package.c

```diff
@@ -975,6 +975,64 @@ int apk_ipkg_add_script(struct apk_installed_package *ipkg,
 	return 0;
 }
 
+static void apk_run_script(const char *script, int dirfd)
+
+{
+	int status;
+	pid_t pid;
+	char *script_args[] = { "/bin/sh", "-c", "", NULL };
+	static char * const environment[] = {
+		"PATH=/usr/sbin:/usr/bin:/sbin:/bin",
+		NULL
+	};
+	script_args[2] = (char *) script;
+
+
+	pid = fork();
+	if (pid == -1)
+		goto error;
+	if (pid == 0) {
+		if (fchdir(dirfd) == 0) {
+			execve(script_args[0], script_args, environment);
+			apk_error("execve() failed for script %s with error str: %s", script, apk_error_str(errno));
+		} else
+			apk_error("change working dir failed with error str: %s", apk_error_str(errno));
+		exit(1);
+	}
+	waitpid(pid, &status, 0);
+	if (!WIFEXITED(status) || WEXITSTATUS(status) != 0) {
+		apk_error("%s: script exited with error %d", script, WEXITSTATUS(status));
+		exit(1);
+	}
+	return;
+error:
+	apk_error("%s: failed to execute: %s", script, apk_error_str(errno));
+}
+
+int apk_run_pre_script(void *ctx, int dirfd, const char *file)
+{
+	char fn[PATH_MAX];
+	if ((apk_flags & (APK_NO_SCRIPTS | APK_SIMULATE)) != 0)
+		return 0;
+	snprintf(fn, sizeof(fn), "./" "%s", file);
+	if (apk_verbosity >= 2)
+		apk_message("Calling apk pre-script: %s\n", fn);
+	apk_run_script(fn, dirfd);
+	return 0;
+}
+
+int apk_run_post_script(void *ctx, int dirfd, const char *file)
```

Comment:
This is basically copy of apk_run_pre_script. Please merge them, use ctx to differentiate the case if needed. If you want to keep database state there, you can add a new field in the apk_database that contains information of commit progress.